src/roman_datamodels/_typing/base.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+import sys
+from typing import Generic, TypeVar
+
+import astropy.units as u
+import numpy as np
+
+if sys.version_info < (3, 10):
+    raise ImportError("Python 3.10+ is required to use the typing module features")
```

Comment:
Yes this fail as soon as it is imported under 3.9 because it is using syntax that is not present in python 3.9. That error is less clear than us raising an error at this point.

Since this module would only be imported during testing or followed by outside code tools I see no issue with not supporting python 3.9 here. Since the completion engines would simply not be able to generate completions.